src/systems/jumpsystem.cpp

```diff
@@ -48,6 +48,7 @@ void JumpSystem::update(float delta)
                 if (entity.hasComponent<Coolant>())
                     jump.addHeat(jump.heat_per_jump);
 
+                jump.charge -= jump.distance;
```

Comment:
This moves the jump charge consumption to after a jump is successful, which allows the aborted jump to consume a different amount. Another solution might be to consume `jump.charge` `onUpdate` as we do `jump.delay`.